### PR TITLE
feat: add feature to detect runtime generated classes

### LINKED ISSUE #61: Explore the SYNTHETIC flag for runtime classes
https://docs.oracle.com/javase/specs/jvms/se17/html/jvms-4.html

```java
import org.objectweb.asm.ClassReader;
import org.objectweb.asm.Opcodes;

import java.io.IOException;
import java.io.InputStream;

public class SyntheticChecker {
    public static boolean isSyntheticClass(InputStream inputStream) throws IOException {
        ClassReader classReader = new ClassReader(inputStream);
        return (classReader.getAccess() & Opcodes.ACC_SYNTHETIC) != 0;
    }
}
```
Goal: add an option to ignore synthetic classes

### PR DESCRIPTION
**Pull request by algomaster99**
Fixes #61

### CODE REVIEW

**Comment by algomaster99** [comment]
Running our tool on sorald-0.8.5 does not produce exit code `1` anymore. See: [`eb3ea91` (#64)](https://github.com/ASSERT-KTH/terminator/pull/64/commits/eb3ea9163f54ae153b7d131244f1fb3297ca90d1)

File: watchdog-agent/src/main/java/io/github/algomaster99/Terminator.java [hunk 1]
```diff
@@ -33,6 +34,9 @@ public byte[] transform(
 
     private static byte[] isLoadedClassWhitelisted(String className, byte[] classfileBuffer) {
         Map<String, List<Provenance>> fingerprints = options.getFingerprints();
+        if (RuntimeClass.isProxyClass(classfileBuffer) || RuntimeClass.isSynthetic(classfileBuffer)) {
```
Comment (MartinWitt) [inline watchdog-agent/src/main/java/io/github/algomaster99/Terminator.java:37]: I am really unsure about synthetic classes. We should keep an eye on them and include them in the threat model.

**Comment by MartinWitt** [comment]
LGTM 👍

File: watchdog-agent/src/main/java/io/github/algomaster99/Terminator.java (hunk 1 above)
Comment (algomaster99) [reply watchdog-agent/src/main/java/io/github/algomaster99/Terminator.java:37]: I would agree with you. We could remove this condition also, as I have never encountered a case yet. The test is also very artificial.

**Pull request merged**